### PR TITLE
Fixed #32004 -- Clarified docs for when request.FILES is set.

### PR DESCRIPTION
**Pull request by janosroden**
Relevant ticket: https://code.djangoproject.com/ticket/32004

Let's discuss how to phrase this.

### CODE REVIEW

**Comment by janosroden** [comment]
Well, CI build failing, its output is missing but I guess because 'behaviour' and 'subpart'. Leave it alone until we discussing.

So @carltongibson, what is the problem exactly?

**Comment by knyghty** [comment]
@janosroden FYI, you can check locally by running `make spelling` from the `docs` directory and checking the output file.

**Comment by felixxm** [comment]
@janosroden Misspelled words are now visible in logs.

**Comment by janosroden** [comment]
@knyghty, @felixxm thank you guys!

**Comment by smithdc1** [comment]
I'm wondering if it is possible to summarise this into a minimal code example. 

Is this possible here? Does a 'code snippet' say a thousand words?

**Comment by janosroden** [comment]
@smithdc1 Do you mean something like [this](https://stackoverflow.com/a/22567429/1396416)? `requests` lib does this automatically.
Even [curl](https://curl.haxx.se/libcurl/c/curl_mime_filedata.html) is too high level :)

Actually it's weird how I fell into this problem... I used a lib which has a class called [MultipartFile](https://pub.dev/documentation/http/latest/http/MultipartFile-class.html) but its `filename` parameter is optional. Pretty misleading.

Do you think is it really worth the effort to extend the documentation? It seems not so easy to post a bad request..

**Comment by carltongibson** [comment]
Hey @janosroden — that was my initial 😬 — Maybe we make it _less clear_... However, let's give it a think. 
Thanks for making the PR!

File: docs/topics/http/file-uploads.txt [hunk 1]
```diff
@@ -38,9 +38,15 @@ subclass) in the form. So the data from the above form would
...
-``request.FILES`` will be empty.
+contain data from form fields if the request method was ``POST`` and
+the ``<form>`` that posted the request has the attribute
+``enctype="multipart/form-data"`` and contains at least one
+``<intput type="file">`` field.
+Or more technically if request method was ``POST``, the
+``Content-Type`` general header was ``multipart/form-data`` and
+the ``Content-Disposition`` subpart header contains the filename parameter
+(default behaviour in browsers for this ``enctype``).
+Otherwise, ``request.FILES`` will be empty.
```
Comment (carltongibson) [inline docs/topics/http/file-uploads.txt:49]: I think I might trim it down to maybe just this: 

```suggestion
contain data from form fields if the request method was ``POST``, at least one
file field was actually posted, and the ``<form>`` that posted the request has
the attribute ``enctype="multipart/form-data"``.
```

I think, in this context, the whole _more technically_ bit is just too much.

**Review by carltongibson — COMMENTED** [review]
Hi @janosroden. 

I've suggested a possible smaller change. What do you think? Would it cover it? 

I'm kind of half-way between something like this and not doing anything at all. 🤔

Thanks again.

File: docs/topics/http/file-uploads.txt (hunk 1 above)
Comment (janosroden) [reply docs/topics/http/file-uploads.txt:49]: I'm fine with it, it's definitely more precise than the current version.
Thanks for your help and time!

Cheers

File: docs/topics/http/file-uploads.txt [hunk 2]
```diff
@@ -38,9 +38,9 @@ subclass) in the form. So the data from the above form would
 be accessible as ``request.FILES['file']``.
 
 Note that :attr:`request.FILES <django.http.HttpRequest.FILES>` will only
-contain data if the request method was ``POST`` and the ``<form>`` that posted
-the request has the attribute ``enctype="multipart/form-data"``. Otherwise,
-``request.FILES`` will be empty.
+contain data from form fields if the request method was ``POST``, at least one
+file field was actually posted, and the ``<form>`` that posted the request has
```
Comment (smithdc1) [inline docs/topics/http/file-uploads.txt:42]: I notice Oxford commas used quite a bit in the docs. Is there a reason for this? Personally, I'm not sure they aid readability in scenarios such as this. 

An example where I think they are helpful: 'I went shopping at marks and spencers, and tesco'. 

I guess what I'm really asking is this a conscious 'house style'? If so I'll bear in mind for the future. :-)

Comment (carltongibson) [reply docs/topics/http/file-uploads.txt:42]: Yep, always use them. (Sorry 🙂)

Comment (felixxm) [reply docs/topics/http/file-uploads.txt:42]: We :heart: them :wink:

Comment (smithdc1) [reply docs/topics/http/file-uploads.txt:42]: Thanks both :-1

**Comment by janosroden** [comment]
Are we done? Thanks for your help!

**Review by carltongibson — APPROVED** [review]
Hi @janosroden. Thanks for this. I've rebased and pushed some minor edits. 

Welcome aboard! ⛵️

**Pull request merged**